traitsui/qt4/tree_editor.py

```diff
@@ -1933,3 +1914,93 @@ def _get_action(self, event):
                 action = None
 
         return (action, to_node, to_object, to_index, data)
+
+
+class TreeItemDelegate(QtGui.QStyledItemDelegate):
+    """ A delegate class to draw wrapped text labels """
+    # FIXME: sizeHint() should return the size required by the label,
+    # which is dependent on the width available, which is different for
+    # each item due to the nested tree structure. However the option.rect
+    # argument available to the sizeHint() is invalid (width=-1) so as a
+    # hack sizeHintChanged is emitted in paint() and the size of drawn
+    # text is returned, as paint() gets a valid option.rect argument.
+
+    # TODO: add ability to override editor in item delegate
+
+    def __init__(self, *args, **kwargs):
+        self.size_map = collections.defaultdict(lambda: QtCore.QSize(1, 41))
+        QtGui.QStyledItemDelegate.__init__(self, *args, **kwargs)
+
+    def sizeHint(self, option, index):
+        """ returns area taken by the text. """
+        return self.size_map[self.editor._tree.itemFromIndex(index)]
+
+    def updateEditorGeometry(self, editor, option, index):
+        """ Update the editor's geometry.
+        """
+        editor.setGeometry(option.rect)
+
+    def paint(self, painter, option, index):
+        """ Render the contents of the item. """
+        item = self.editor._tree.itemFromIndex(index)
+        expanded, node, instance = self.editor._get_node_data(item)
+        column = index.column()
+
+        renderer = node.get_renderer(object, column=column)
+        if renderer is None and self.editor.factory.word_wrap:
+            renderer = WordWrapRenderer()
```

Comment:
I don't think you want to do this here. You're in a pretty hot code path (I think)